contracts/LosslessReporting.sol

```diff
@@ -302,11 +302,15 @@ contract LosslessReporting is ILssReporting, Initializable, ContextUpgradeable,
     /// @param _token Address of the token to be retrieved
     /// @param _to Address to which the tokens will be sent
     /// @param _amount Amount of tokens to be retrieved
-    function retrieveRejectedReportTokens(ILERC20 _token, address _to, uint256 _amount) public onlyLosslessAdmin {
+    /// @param _reportId ID of the associated report to check its status
+    function retrieveRejectedReportTokens(ILERC20 _token, address _to, uint256 _amount, uint256 _reportId) public onlyLosslessAdmin {
         require(_to != address(0), "LSS: Cannot send to zero address");
-        require(_token != stakingToken, "LSS: Cannot retrieve staking tokens");
         require(_amount > 0, "LSS: Amount must be greater than zero");
 
+        // Check the status of the report
+        Report storage report = reportInfo[_reportId];
+        require(report.status == ReportStatus.Rejected || (report.status == ReportStatus.Expired && block.timestamp > report.reportTimestamps + reportLifetime), "LSS: Report not expired or rejected");
+
         uint256 contractBalance = _token.balanceOf(address(this));
```

Comment:
But this allows us to withdraw all the staking tokens when any report is expired. So if report x is still active, but we submit another id which is expired we can drain all the funds. So this should transfer our only the reporting amount.

Reply:
Yeah, that's true. My bad I went int with a more broader approach but I understand it needed to be specific now. Did the necessary refactors. [5d8ff3e](https://github.com/Lossless-Cash/lossless-v3/pull/116/commits/5d8ff3ed7d3f01b060364345465f1b87cf69987a)